### PR TITLE
feat: add IconButton

### LINKED ISSUE #169: Component: Button (copy)
### Design
https://www.figma.com/file/FWlI63N28LP7BxqNqj1bGw/sy-004-(Design!)?node-id=3708%3A14588

### Things to consider
- Take the "Hover" rules into account: https://www.figma.com/file/FWlI63N28LP7BxqNqj1bGw/sy-004-(Design!)?node-id=122%3A1191

### LINKED ISSUE #228: Button Issues
Our button is based on a real button element. That's good.

The implementation looks odd though:
https://github.com/satellytes/satellytes.com/blob/07e2040d4e548c0a6d4c368b1369b5a8acd9ab95/src/components/buttons/button.tsx#L29-L37

You can see that there is a dedicated `onClick` event handler whereas I would have expected to spread the props to pass through any option that is valid for a button (disabled, onClick and such).

Something like that:

```
export interface ButtonProps extends React.ComponentPropsWithoutRef<"button"> {
// additional props if any
}

const Button = (props: ButtonProps) => {
  return (
    <StyledButton {...props}>
      <ButtonText>{props.children}</ButtonText>
```

This is a much nicer citizen of a UI component library, isn't it ?

### PR DESCRIPTION
**Pull request by milljoniaer**
### Description
This adds an IconButton, which displays Icons which should be passed as children (Icon components).

Additionally this wraps the stories of the Regular Button and the IconButton into one.

image

---

closes #169

### CODE REVIEW

**Comment by feedm3** [comment]
You can make this a lot more reuseable, by not creating a `CopyButton` (I would not call this `Copy`, as this name is also used for other components like text stuff) but a generic  `IconButton`. Because you `CopyButton` is just an `IconButton` with the `navigator.clipboard.writeText(text);` click handler. The copy can be a good example in Storybook ;-) 

Look how material UI is doing the implementation ([Docs](https://mui.com/components/buttons/#upload-button)), maybe you can do it like that:
```
<IconButton aria-label="upload picture">
  <PhotoCamera />
</IconButton>
```

File: src/components/buttons/copy.stories.tsx
Comment (georgiee) [inline src/components/buttons/copy.stories.tsx]: I know this is fun for yourself but please don't leak informal palceholders like that into the sources. You should use lorem ipsum or some descriptive text instead.

**Review by georgiee — CHANGES_REQUESTED** [review]
My first feeling also was: Oh gosh, this is a missed opportunity for an icon button (like Fabian commented) but I know where you are coming from. We have an issue for exactly this button type so you made the right conclusions: issue -> new component -> new story.

Variant 1:
Here I would appreciate if you remove the story and just include the button in the existing button story. In addition we need a little bit more to make this a useful component. Foremost the copy feedback. Otherwise this component really don't bring much.

See the linked issue about [copy feedback](https://github.com/satellytes/satellytes.com/issues/136) and maybe ask the design team for guidance.


Variant 2:
Postpone this PR and deliver a truly icon component. Combined with the new icon component I'm offering through a PR (#230) this should be not that complex I hope.

I think variant 2 can be quicker but variant 1 could be fun with some small animation combined with useSprings and such. Your choice and I put this PR on hold until then. 

I want to emphasize: This is by no means a mistake by you but a situation where the lack of requirements causes some additional thoughts and adjustment the moment you opened this specific task. 

Thank you!💪

**Comment by milljoniaer** [comment]
Thank you for comments. Together with @fischeversenker, I have decided to change the Copy component into an IconButton component as @feedm3 suggested.

@georgiee I would suggest to do the onClick animation of Icon Buttons or Buttons in general another Issue iteration and stick to the basic design requirements for now, if that is okay?

Please have another look at the changes I have made.

File: src/components/buttons/button.stories.mdx
Comment (fischeversenker) [inline src/components/buttons/button.stories.mdx]: > Only the valid types of those property differ for the different components.

I don't understand. What do you mean by that?

File: src/components/buttons/button.stories.mdx
Comment (fischeversenker) [inline src/components/buttons/button.stories.mdx]: Well done, providing a link to the `Icon` component 👌

File: src/components/buttons/button.stories.mdx
Comment (fischeversenker) [inline src/components/buttons/button.stories.mdx]: The empty `children` property is confusing here. Looks like a leftover?

Also the indentation in this story leads to weird indentation in Storybook:
image

I couldn't find a way to get rid of this behavior except to write everything in one line. Let's ignore this for now, but let's at least remove the empty `children` property:
```suggestion
    <Button onClick={() => console.log('Regular Button clicked')}>
      Alle Services
    </Button>
```

**Review by fischeversenker — COMMENTED** [review]
Thanks for the recent changes and the additional documentation for the existing `Button` 👌 
I left a few comments. Especially the one regarding the first paragraph of the documentation is something I'd like to clarify before we merge this.

Also, one more thing that would be nice to have in this PR would be move both of these new components into the `new-components` folder. It should be as easy as moving these three files since they are not yet used anywhere.

File: src/components/buttons/button.stories.mdx
Comment (feedm3) [reply src/components/buttons/button.stories.mdx]: Also note that `children` is a special property in React that should never be set as an attribute.

File: src/components/buttons/icon-button.tsx
Comment (georgiee) [inline src/components/buttons/icon-button.tsx]: You only expect to receive an icon through children. You never expect more. Why would you use the children? There is no rule to use children so I would omit it. Instead I would try a similar technique as with the illustrations:

1. Hard code the icon here and only pass in the icon name
2. The prop for the icon can be typed similar to how the accordion types the illlustration type. Great for auto completion and icon safety.

See the accordion:

https://github.com/satellytes/satellytes.com/blob/5cc2990d9777d0791d165ae8ca12570fcf59ab0d/src/components/accordion/accordion.tsx#L36

File: src/components/buttons/icon-button.tsx
Comment (georgiee) [inline src/components/buttons/icon-button.tsx]: I would not expose the `onClick` instead see  this general issue:
https://github.com/satellytes/satellytes.com/issues/228

Passing in `onClick` feels super weird for things that support that by default. It's always a big warning sign so please be careful with that in the future.

**Review by georgiee — CHANGES_REQUESTED** [review]
I have left a comment about the props mainly which I would like to see addressed or discussed. Thanks.

File: src/components/buttons/icon-button.tsx
Comment (fischeversenker) [reply src/components/buttons/icon-button.tsx]: To make it more clear: The idea is to use `export interface CopyProps extends React.ComponentPropsWithoutRef<"button">` and then to spread the `props` (by the way, they should really be called `IconButtonProps` instead of `CopyProps`).

I don't think of this as such a big deal. The component receives an `onClick` and passes it on to the native `button` element. What's wrong with that approach? It's very explicit but other then that I dont't see any flaws. To the outside (users of this component) it doesn't make a difference, does it?

Comment (georgiee) [reply src/components/buttons/icon-button.tsx]: Yes it works, but it's not idiomatic to React. That's the deal. I could ask instead why not use `type CopyProps = any`, spread the props and say that people don't need the types because buttons always have `onClick`. Still works and has no flaws.

Instead I went the route following the framework's rails and suggested to use the given button type that helps us to map react components back to the web standard so you can use the `<IconButton/>` as you would use any other `<button/>`. This resonates so much more with me then any custom mapping of `onClick` which creates more noise then the special type `ComponentPropsWithoutRef` I referenced before.

File: src/components/buttons/button.stories.mdx
Comment (milljoniaer) [reply src/components/buttons/button.stories.mdx]: Due to some functionality changes, I have left out the whole sentence.

File: src/components/buttons/button.stories.mdx
Comment (milljoniaer) [reply src/components/buttons/button.stories.mdx]: I had that in the previous story to let the user change the label of the button in the canvas. It was a leftover. I fixed it.

File: src/components/buttons/icon-button.tsx
Comment (milljoniaer) [reply src/components/buttons/icon-button.tsx]: Thank you for your suggestions, I do like @georgiee idea and changed it, just like he suggested it. I hope that is fine for everyone.

File: src/components/buttons/icon-button.tsx
Comment (milljoniaer) [reply src/components/buttons/icon-button.tsx]: Nice idea, way cleaner, I have done it like that 👍

**Comment by milljoniaer** [comment]
Thank you for all your suggestions. I have now changed the following:
- I moved the whole `buttons` folder into the `new-components` folder
- The IconButton now passes a `show` property to the hardcoded Icon Component 
- I removed the unnecessary children property everywhere (especially in the story)
- I am now using `React.ComponentPropsWithoutRef<'button'>` and spread the props into the `<button>` 

Please have another look.

**Comment by milljoniaer** [comment]
To make both buttons equal in use, I let also the RegularButtonProps extend React.ComponentPropsWithoutRef<'button'> and spreaded them over the native <button>.

I am not sure if that is the right PR for this. But in that way, both buttons are immediately equal in use.

File: src/new-components/buttons/icon-button.tsx
Comment (georgiee) [inline src/new-components/buttons/icon-button.tsx]: please use `icon`. We use `show` on the icon because it feels awkward to use `<Icon icon="..."`.

File: src/new-components/buttons/button.stories.mdx
Comment (georgiee) [inline src/new-components/buttons/button.stories.mdx]: Why h3 + h4 ? I would just go straight to `## Regular`

**Review by georgiee — APPROVED** [review]
Thanks, looks good. Let's go with that. I have some small little things. Please adjust but you get my approval already 🙌

**Pull request merged**
